StRoot/StFcsWaveformFitMaker/StFcsWaveformFitMaker.cxx

```diff
@@ -1066,27 +1105,94 @@ void StFcsWaveformFitMaker::drawCh(UInt_t detid, UInt_t ch) const
+	    for( int i=0; i<ana->NPeaks(); ++i ){
+	      if( (mCenterTB-prexing) < ana->GetPeak(i).mPeakX && ana->GetPeak(i).mPeakX < (mCenterTB+postxing) ){
+		if( !foundtrigidx && trigfitidx==i ){ foundtrigidx = true; trigfitidx = validpeakidx.size(); }//do before adding i to vector so that when i does get added the triggered crossing index matches
+		if( !validpeakidx.empty() && validpeakidx.back()!=(i-1) ){ LOG_ERROR << "StFcsWaveformFitMaker::dualFit: Found indices are not linear" << endm;  }
+		validpeakidx.push_back(i);
+		if( xmin > ana->GetPeak(i).mStartX ){ xmin = ana->GetPeak(i).mStartX; }
+		if( xmax < ana->GetPeak(i).mEndX ){ xmax = ana->GetPeak(i).mEndX; }
+	      }
+	    }
+	    if( !foundtrigidx ){ LOG_ERROR << "StFcsWaveformFitMaker::dualFit: Unable to find a matching triggered crossing possibly due to improper use of function" << endm; }
+	    //End of Hack
+	    npeaks = validpeakidx.size();
+	    if( npeaks>0 && npeaks<mMaxPeak ){
+	      TF1* func_PulseFit = mDbPulse->createPulse(xmin,xmax,2+npeaks*3);//only fit inside the range of valid peaks
+	      ana->SetFitPars(func_PulseFit,validpeakidx.front(),validpeakidx.back());
+	      if( func_PulseFit->GetParameter(0)==0 ){ LOG_ERROR << "StFcsWaveformFitMaker::drawDualFit: Unable to set function parameters, bad peak indicies" << endl; }	      
+	      //ana->SetSignal(func);
+	      TFitResultPtr result = ggdraw->Fit(func_PulseFit,"BNRQ");
+	      //compidx no longer equal to index to triggered crossing peak
+	      func_PulseFit->SetLineColor(kBlack);
+	      func_PulseFit->SetLineWidth(1);
+	      func_PulseFit->SetBit(kCanDelete);
+	      func_PulseFit->Draw("same");
+	    }
+	  }
+	}
+	TF1* func_gaus = 0;
+	float gausres[8] = {0};
+	gausFit( ggdraw , gausres, func_gaus );
+	if( func_gaus!=0 ){
+	  func_gaus->SetLineColor(kGreen);
+	  func_gaus->SetLineWidth(1);
+	  func_gaus->SetBit(kCanDelete);
+	  func_gaus->SetParameter(1,0.5);
+	  func_gaus->Draw("same");
+	}
+      }//ch check
+    }//detid check
+  }//hit loop
 }
 
 float StFcsWaveformFitMaker::gausFitWithPed(TGraphAsymmErrors* g, float* res, TF1*& func){
-  int n = g->GetN();
-  double *t = g->GetX();
-  double *a = g->GetY();    
-  int p=0;
-  double sumsq = 0;//For variance
-  for(int i=0; i<n; i++){
-    int tb1=t[i];
-    if(tb1>=mPedMin && tb1<=mPedMax){    
-      p+=a[i];
-      sumsq += a[i]*a[i];
-    }
-  }
-  float ped = float(p)/(mPedMax-mPedMin+1.0);
-  res[6] = ped;
-  res[7] = sqrt( ( sumsq-((double(p)*double(p))/(mPedMax-mPedMin+1.0)) )/(mPedMax-mPedMin) );//Variance/StdDev using naive algorithm from https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
-  printf("Pedestal=%6.2f/(%6.2f-%6.2f+1)=%6.2f +- %6.2f\n",float(p),float(mPedMax),float(mPedMin),ped,res[7]);
-  return gausFit(g, res, func, ped);
+  AnaPed( g, res[6], res[7] );
+  printf("Pedestal (%6.2f-%6.2f+1)=%6.2f +- %6.2f\n",float(mPedMax),float(mPedMin),res[6],res[7]);
```

Comment:
Another printf should be changed to LOG_INFO. It is kind of important if this maker is to be used in production

Reply:
Changed to LOG_DEBUG